configure.ac

```diff
@@ -352,18 +352,47 @@ AC_ARG_VAR([POLARSSL_CFLAGS], [C compiler flags for PolarSSL])
 AC_ARG_VAR([POLARSSL_LIBS], [linker flags for PolarSSL])
 if test -z "${POLARSSL_LIBS}"; then
 	AC_CHECK_LIB(
-		[polarssl],
-		[x509parse_crt],
+		[mbedtls],
+		[mbedtls_x509_crt_parse],
 		[
-			POLARSSL_LIBS="-lpolarssl"
+			POLARSSL_LIBS="-lmbedtls -lmbedx509 -lmbedcrypto"
 			have_polarssl="yes"
 		],
-		[have_polarssl="no"]
+		AC_CHECK_LIB(
+			[polarssl],
+			[x509_crt_parse],
+			[
+				POLARSSL_LIBS="-lpolarssl"
+				have_polarssl="yes"
+			],
+			[have_polarssl="no"]
+		)
 	)
 else
 	have_polarssl="yes"
 fi
 
+if test "${have_polarssl}" = "yes"; then
+    old_CFLAGS="${CFLAGS}"
+	old_LIBS="${LIBS}"
+    CFLAGS="${CFLAGS} ${POLARSSL_CFLAGS}"
+	LIBS="${LIBS} ${POLARSSL_LIBS}"
+	AC_CHECK_FUNC(
+		[mbedtls_x509_crt_parse],
+		[
+			have_mbedtls="yes"
+            AC_DEFINE([HAVE_LIBMBEDTLS], [1], [Using mbed TLS 2.x])
+		],
+		[
+			have_mbedtls="no"
+		]
+	)
```

Comment:
I can put it on one line if you want.  I just liked the symmetry better this way.

The 1.x versions are called 'polarssl', with newer 1.3 versions partly rebranded to mbedtls.  Include path however remains polarssl/, and we keep linking to -lpolarssl.

Only in the 2.x branch they completed the renaming.  Hence the distinction.